gatewayconfig/gatewayconfig_app.py

```diff
@@ -80,15 +78,24 @@ def init_sentry(self, sentry_dsn, balena_app_name, balena_device_uuid, variant):
     def init_nmcli(self):
         nmcli_custom.disable_use_sudo()
 
-    def init_gpio(self):
+    def init_gpio(self, variant):
+        """
+        This code was originally written for Raspberry Pi but ROCK Pi does not
+        support gpiozero. Custom GPIO implementations for ROCK Pi are used based
+        on the detected hardware.
+        """
         if is_raspberry_pi():
             self.user_button = Button(self.get_button_pin(), hold_time=USER_BUTTON_HOLD_SECONDS)
-            self.user_button.when_held= self.start_bluetooth_advertisement
             self.status_led = LED(self.get_status_led_pin())
+        elif is_rockpi():
+            Device.pin_factory = MockFactory()
+            self.user_button = MraaButton(16, hold_seconds=USER_BUTTON_HOLD_SECONDS)
+            self.user_button.start()
+            self.status_led = MraaLED(18)
         else:
-            logger.warn("LEDs and buttons are disabled. GPIO not yet supported on this device.")
-            self.user_button = None
-            self.status_led = None
+            raise UnsupportedGpioException("GPIO is not supported for %s" % variant)
```

Comment:
On second thought, the existing `else` behavior is probably better. That way you can still use the app, just no button or LED.

Reply:
Reverted behavior: https://github.com/NebraLtd/hm-config/pull/158/files#diff-8f3c7cad01ea4ab5283fb98a71afe99f7f4a170a2387b7477d6755990a11f577R102